base/glibc-compatibility/spawn.h

```diff
@@ -6,11 +6,16 @@ extern "C" {
 #endif
 
 #include <features.h>
+#include <sys/types.h>
+#include <bits/types/sigset_t.h>
+
+#define POSIX_SPAWN_SETSIGMASK		0x08
 
 typedef struct {
 	int __flags;
 	pid_t __pgrp;
-	sigset_t __def, __mask;
+	sigset_t __sd;
+	sigset_t __ss;
```

Comment:
Could you please elaborate on changes in this file?

Reply:
The `posix_spawnattr_t` that is defined in this header is differs from the system wide impementation it has `__fn` (used as part of padding, this is OK), that is set in `glibc-compatbility.c` in `posix_spawnp`, and it calls `posix_spawn`, but, `base/glibc-compatibility/musl/posix_spawn.c` knows nothing about this field.

So, I need to match all used fields with the system wide version to make it common

And due to this and too small stack `posix_spawnp` (`p` stands for search in `PATH`) never worked, while `chdig` uses it, so I fixed it.